### PR TITLE
[SPARK-8965][Docs] Add ml-guide Python Example: Estimator, Transformer, and Param

### PR DESCRIPTION
**Pull request by Rosstin**
Added ml-guide Python Example: Estimator, Transformer, and Param
/docs/_site/ml-guide.html

### CODE REVIEW

File: docs/ml-guide.md [hunk 1]
```diff
@@ -355,6 +355,75 @@ jsc.stop();
...
+from pyspark.mllib.regression import LabeledPoint
+from pyspark.ml.classification import LogisticRegression
+from pyspark.ml.param import Param, Params
+from pyspark.sql import Row, SQLContext
+
+sc = SparkContext(appName="SimpleParamsExample")
+sqlContext = SQLContext(sc)
+
+# Prepare training data.
+# We use LabeledPoint.
```
Comment (jkbradley) [inline docs/ml-guide.md]: "Spark SQL can convert RDDs of LabeledPoints into DataFrames."

File: docs/ml-guide.md [hunk 2]
```diff
@@ -355,6 +355,75 @@ jsc.stop();
...
+training = sc.parallelize(
+[LabeledPoint(1.0, [0.0, 1.1,  0.1]),
```
Comment (jkbradley) [inline docs/ml-guide.md]: Use PEP8 style guidelines for Python: [http://legacy.python.org/dev/peps/pep-0008/]
(The issue here is indentation.)

File: docs/ml-guide.md [hunk 3]
```diff
@@ -355,6 +355,75 @@ jsc.stop();
...
+print "Model 1 was fit using parameters: "
+print model1.extractParamMap()
+
+# We may alternatively specify parameters using a Python dictionary as a paramMap
+paramMap = {lr.maxIter: 20}
+paramMap[lr.maxIter] = 30 # Specify 1 Param, overwriting the original maxIter.
+paramMap.update({lr.regParam: 0.1, lr.threshold: 0.55}) # Specify multiple Params.
+
+# You can combine paramMaps, which are python dictionaries.
+paramMap2 = { lr.probabilityCol: "myProbability" } # Change output column name
```
Comment (jkbradley) [inline docs/ml-guide.md]: python style: remove spaces: ```{lr.probabilityCol: "myProbability"}```

File: docs/ml-guide.md [hunk 4]
```diff
@@ -355,6 +355,75 @@ jsc.stop();
...
+paramMapCombined = paramMap.copy()
+paramMapCombined.update(paramMap2)
+
+# Now learn a new model using the paramMapCombined parameters.
+# paramMapCombined overrides all paramters set earlier via lr.set* methods.
```
Comment (jkbradley) [inline docs/ml-guide.md]: typo: paramters

File: docs/ml-guide.md [hunk 5]
```diff
@@ -355,6 +355,75 @@ jsc.stop();
...
+model2 = lr.fit(training.toDF(), paramMapCombined)
+print "Model 2 was fit using parameters: "
+print model2.extractParamMap()
+
+# Prepare test data
+test = sc.parallelize(
+[LabeledPoint(1.0, [-1.0, 1.5,  1.3]),
```
Comment (jkbradley) [inline docs/ml-guide.md]: fix indentation

File: docs/ml-guide.md [hunk 6]
```diff
@@ -355,6 +355,75 @@ jsc.stop();
...
+LabeledPoint(1.0,  [ 0.0, 2.2, -1.5])])
+
+# Make predictions on test data using the Transformer.transform() method.
+# LogisticRegression.transform will only use the 'features' column.
+# Note that model2.transform() outputs a "myProbability" column instead of the usual
+# 'probability' column since we renamed the lr.probabilityCol parameter previously.
+prediction = model2.transform(test.toDF())
+selected = prediction.select("features", "label", "myProbability", "prediction")
+for row in selected.collect():
+	print row
```
Comment (jkbradley) [inline docs/ml-guide.md]: fix indentation

**Comment by jkbradley** [comment]
@Rosstin Thanks!  Just minor comments.  Btw, this PR will be blocked by [https://github.com/apache/spark/pull/8079], which adds "threshold" back, but that should get merged soon...after flaky tests pass.

**Comment by Rosstin** [comment]
No problem, I'll fix these as soon as I get the chance.

On Mon, Aug 10, 2015 at 6:06 PM, jkbradley <notifications@github.com> wrote:

> @Rosstin <https://github.com/Rosstin> Thanks! Just minor comments. Btw,
> this PR will be blocked by [#8079
> <https://github.com/apache/spark/pull/8079>], which adds "threshold"
> back, but that should get merged soon...after flaky tests pass.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/8081#issuecomment-129661310>.
>



-- 
Road Redemption <http://store.steampowered.com/app/300380/> • Rex Rocket
<http://store.steampowered.com/app/288020/> • Queen At Arms
<https://www.facebook.com/QueenAtArms> • Hilton HHonors Straight-to-Room App
<http://news.hiltonworldwide.com/index.cfm/newsroom/detail/27192>

**Comment by Rosstin** [comment]
Let me know if this covers it! :-)

**Comment by jkbradley** [comment]
ok to test

**Comment by jkbradley** [comment]
Jenkins test this please

**Comment by jkbradley** [comment]
LGTM pending tests.  Thanks!

**Comment by jkbradley** [comment]
I'll merge this with master and branch-1.5.  Thanks!

**Pull request closed without merging**